primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js

```diff
@@ -195,26 +186,8 @@ PrimeFaces.widget.FileUpload = PrimeFaces.widget.BaseWidget.extend({
             add: function(e, data) {
                 $this.chooseButton.removeClass('ui-state-hover ui-state-focus');
 
-                if($this.fileAddIndex === 0) {
-                    $this.clearMessages();
-                }
-
-                // we need to fake the filelimit as the jquery-fileupload input always only contains 1 file
-                var fileLimit = data.fileInput.data('p-filelimit');
```

Comment:
Did you test filelimit with advanced fileupload?

Reply:
Yep and it it stopped it at 2 when I tried to upload 3 files and the message was correct.

Reply:
![image](https://github.com/primefaces/primefaces/assets/4399574/1e00e07c-b9a2-4093-b1b8-6aa00a86414c)